src/opm/parser/eclipse/EclipseState/Schedule/MSW/WellSegments.cpp

```diff
@@ -545,6 +545,23 @@ namespace Opm {
         return true;
     }
 
+    bool WellSegments::updateWSEGAICD(const std::vector<std::pair<int, AutoICD> >& aicd_pairs) {
```

Comment:
Here you could pass a `KeywordLocation` argument, so that you can write a better error message (with filename and linenr).

Reply:
It looks like this can only refer to the line number of the `Keyword`, but not the `Record`, right?

Reply:
That is true; we only support the line number of the keyword start; better than nothing though :-)